### ISSUE TITLE
Basis.is_equal_approx fails even when comparing identical basises

### ISSUE THREAD

**Issue by Demindiro** [godotengine/godot#45062]
Probably related: #37390

**Godot version:**
`3.2.3.stable.custom_build.31d0f8ad8`

**OS/device including version:**
Linux pc 4.19.0-12-amd64 #1 SMP Debian 4.19.152-1 (2020-10-18) x86_64 GNU/Linux

**Issue description:**
`is_equal_approx` returns false when comparing the same Basis (only happens in GDScript)

The issue, I think, is that `VCALL_PTR2R(Basis, is_equal_approx);` should actually be `VCALL_PTR1R(Basis, is_equal_approx);`. Now it's calling the 2 parameter version when it should be calling the one parameter version.

https://github.com/godotengine/godot/blob/ecf432abc7b3cd68c618150c63430323913c47ee/core/variant_call.cpp#L849

**Steps to reproduce:**
```gdscript
extends MainLoop

func _idle(d):
	var b := Basis(Vector3.ONE, Vector3.ONE, Vector3.ONE)
	printt(b.is_equal_approx(b))
	return true
```
Also "works" with "valid" rotation basises.

**Minimal reproduction project:**
See above

**Comment by aaronfranke** [comment]
Test code:

```gdscript
	var b1 = Basis(Vector3(1, 0, 0), Vector3(0, 1, 0), Vector3(0, 0, 1))
	print(b1.is_equal_approx(b1))
	var b2 = Basis(Vector3(2, 0, 0), Vector3(0, 2, 0), Vector3(0, 0, 2))
	print(b2.is_equal_approx(b2))
	var b3 = Basis(Vector3(0.70710678118, 0.70710678118, 0), Vector3(0, 1, 0), Vector3(0, 0, 1))
	print(b3.is_equal_approx(b3))
```

Output in master: 

```
True
True
True
```

Output in 3.2:

```
True
False
False
```

It seems that the method in 3.2 only works when the Basis is orthonormalized.

**Comment by aaronfranke** [comment]
Wow, I've been reviewing the history of this for the past hour. [Here is the root of the problem](https://github.com/godotengine/godot/blame/1fed266bf5452b30376db62495f4985f6975f2c1/core/math/basis.cpp#L560), @reduz wrote:

```cpp
bool Basis::is_equal_approx(const Basis &a, const Basis &b, real_t p_epsilon) const {
```

The above is an instance method which doesn't use an instance, which is bogus behavior. It means that you would need to do (in core, since this wasn't ever exposed to GDScript in this form):

```gdscript
b1.is_equal_approx(b2, b3)
```

Which would then discard `b1` and compare `b2` and `b3`.

Then, [this was later bound](https://github.com/godotengine/godot/commit/dee98d3b6d58cbe42fc403d999b278aec9447105#diff-4421d1b614fed4b54265f445a50a3521aeec63ab0efa55a9df639cd00cf987feR1847) by @reduz, a very subtle and innocent mistake caused by the method itself being bogus:

```cpp
ADDFUNC2R(BASIS, BOOL, Basis, is_equal_approx, BASIS, "b", FLOAT, "epsilon", varray(CMP_EPSILON));
```

Since this is actually an instance method, when bound to GDScript and we use `b1.is_equal_approx(b2)`, `b1` is discarded, and then `b2` is passed as `a`. For `b3`, `CMP_EPSILON` is silently auto-converted to `Basis` (which becomes the identity `Basis()`), and then it compares `b2` to the identity.

Now... what should we do about this? What we really want is to compare `b1` and `b2` and discard `b3`. In master we got rid of `b3` by binding the one-arg version, but for 3.2, here is a simple fix that doesn't break ABI compatibility.

```diff
-	bool is_equal_approx(const Basis &a, const Basis &b) const { return a.is_equal_approx(b); }
+	bool is_equal_approx(const Basis &a, const Basis &b) const { return is_equal_approx(a); }
```

Output of the above test code with this change:

```
True
True
True
True
```

**Comment by akien-mga** [comment]
Fixed by #45065.
